arangod/Pregel/Graph.h

```diff
@@ -184,19 +188,20 @@ class Vertex {
     TRI_ASSERT(this->keyLength() == keyLength);
   }
 
-  uint16_t keyLength() const noexcept { return _keyLength; }
+  [[nodiscard]] uint16_t keyLength() const noexcept { return _keyLength; }
 
-  std::string_view key() const { return std::string_view(_key, keyLength()); }
+  [[nodiscard]] std::string_view key() const {
+    return std::string_view(_key, keyLength());
+  }
   V const& data() const& { return _data; }
   V& data() & { return _data; }
 
-  PregelID pregelId() const {
-    return PregelID(_shard, std::string(_key, keyLength()));
+  [[nodiscard]] PregelID pregelId() const {
+    return {_shard, std::string(_key, keyLength())};
```

Comment:
I found the original more clear. If it must be a brace-initialisation (why?), then
```suggestion
    return PregelID{_shard, std::string(_key, keyLength())};
```